src/citrine/_session.py

```diff
@@ -128,6 +128,28 @@ def delete_resource(self, path: str) -> dict:
         """DELETE a particular resource as JSON."""
         return self.checked_delete(path).json()
 
+    @staticmethod
+    def cursor_paged_resource(base_method: Callable[..., dict], path: str, *args,
+                              forward: bool = True, per_page: int = 100, **kwargs):
+        """
+        Returns a flat generator of results for an API query.
+
+        Results are fetched in chunks of size `per_page` and loaded lazily.
+        """
+        params = kwargs.get('params', {})
+        params['forward'] = forward
+        params['ascending'] = forward
```

Comment:
What's the difference between `forward` and `ascending`?

Reply:
https://github.com/CitrineInformatics/platform-backend/blob/develop/code/rest/rest-base/src/main/scala/io/citrine/rest/model/Page.scala talks about this -- the gist is that `ascending` is an absolute sort order for the results, and forward tells the backend whether to return to results "ahead of" or behind the cursor, i.e. for a page size of 2 and a cursor pointing at 5 you would get `3, 4` for `forward = false` and `ascending = true`, `4, 3` for `forward = false` and `ascending = true`, `6, 7` for `forward = true` and `ascending = true`. If you never reverse directions `forward` will always equal `ascending`.

Reply:
Thanks. Sounds like it's fine for the client to always set `params['forward']` and `params['ascending']` to be the same, and that this ability to reverse directions might be more important for a graphical interface.

Reply:
I would say that's accurate, yeah.